packages/react-native-editor/__device-tests__/gutenberg-editor-writing-flow-@canary.test.js

```diff
@@ -0,0 +1,344 @@
+			paragraphBlockElement,
+			testData.shortText
+		);
+
+		const imageBlockElement = await editorPage.getBlockAtPosition(
+			blockNames.image
+		);
+		await imageBlockElement.click();
+		await editorPage.driver.pause( 1000 );
+
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( false );
+		expect( await editorPage.isImageBlockSelected() ).toBe( true );
+	} );
+
+	it( 'should manually dismiss the keyboard', async () => {
+		await editorPage.initializeEditor( { initialTitle: '' } );
+
+		const defaultBlockAppenderElement =
+			await editorPage.getDefaultBlockAppenderElement();
+		await defaultBlockAppenderElement.click();
+
+		const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+			blockNames.paragraph
+		);
+		expect( paragraphBlockElement ).toBeTruthy();
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( true );
+
+		await editorPage.dismissKeyboard();
+
+		// Checks that no block is selected by looking for the block menu actions button
+		expect( await editorPage.isBlockActionsMenuButtonDisplayed() ).toBe(
+			false
+		);
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( false );
+	} );
+
+	it( 'should dismiss the keyboard and show it back when opening modals', async () => {
+		await editorPage.initializeEditor( { initialTitle: '' } );
+
+		const defaultBlockAppenderElement =
+			await editorPage.getDefaultBlockAppenderElement();
+		await defaultBlockAppenderElement.click();
+
+		await editorPage.openBlockSettings();
+		await editorPage.driver.pause( 1000 );
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( false );
+
+		await editorPage.dismissBottomSheet();
+		await editorPage.driver.pause( 1000 );
+
+		expect( await editorPage.driver.isKeyboardShown() ).toBe( true );
+		const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+			blockNames.paragraph
+		);
+		await editorPage.typeTextToTextBlock(
+			paragraphBlockElement,
+			testData.listItem1
+		);
+		const typedText = await paragraphBlockElement.getText();
+		expect( typedText ).toMatch( testData.listItem1 );
+	} );
+
+	it( 'should be able to split one paragraph block into two', async () => {
+		await editorPage.initializeEditor();
+
+		const defaultBlockAppenderElement =
+			await editorPage.getDefaultBlockAppenderElement();
+		await defaultBlockAppenderElement.click();
+
+		const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+			blockNames.paragraph
+		);
+		await editorPage.typeTextToTextBlock(
+			paragraphBlockElement,
+			testData.shortText
+		);
+		await clickMiddleOfElement( editorPage.driver, paragraphBlockElement );
+		await editorPage.typeTextToTextBlock( paragraphBlockElement, '\n' );
+
+		const text0 = await editorPage.getTextForParagraphBlockAtPosition( 1 );
+		const text1 = await editorPage.getTextForParagraphBlockAtPosition( 2 );
+
+		expect( await editorPage.getNumberOfParagraphBlocks() ).toEqual( 2 );
+		expect( text0 ).not.toBe( '' );
+		expect( text1 ).not.toBe( '' );
```

Comment:
In the original test, it was [also checking](https://github.com/WordPress/gutenberg/pull/57460/files#diff-7c0dd43f86cdf2dd97aeef3406049ee2a84c0b4708d82e88fc420b526ad3c234L34) that the two texts were a match with the original text using a Regex with two options, this is a bit flaky because the caret's position to split the initial Paragraph block is not always the same one.

I think it would be enough to check that they're not empty, we do manual testing with this case anways.

Reply:
>I experimented enabling this when using Appium and triggering the arrow keys but doesn't work.

I think it's only supported on Android, to be able to send keys.

> An alternative to check paragraph split is by merging two before. 

What do you mean by that?

Reply:
> > An alternative to check paragraph split is by merging two before.
> 
> What do you mean by that?

I meant that we could test the split action if we merge two paragraphs before testing the split. I.e. do the following:
1. Add paragraph with text 1.
2. Add paragraph with text 2.
3. Select the second paragraph and merge it with the first paragraph.
4. At this point the caret will be positioned in the expected position to split the paragraph.
5. Split the paragraph into two.
6. Assert that each paragraph contains the expected text.

In the end, we could test both merge and split actions in the same test.

Reply:
Good idea! Thanks for the explanation, I like the approach and the suggestion of unifying both tests. I'll work on the changes.

Reply:
Updated in https://github.com/WordPress/gutenberg/pull/57460/commits/b4522c309abab7bc0f23009a98ba72230575fa11 I also added this change https://github.com/WordPress/gutenberg/pull/57460/commits/918be74ee6fb4aba99e194c151b0b4e76775716d needed for this test. Let me know what you think, thank you!

Reply:
It looks great, thanks for applying the suggestion 🙇 !